xm-commons-logging/src/main/java/com/icthh/xm/commons/logging/config/LoggingConfig.java

```diff
@@ -0,0 +1,108 @@
+package com.icthh.xm.commons.logging.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.icthh.xm.commons.logging.util.LogObjectPrinter;
+import lombok.Data;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.icthh.xm.commons.logging.util.LogObjectPrinter.*;
+
+@Data
+public class LoggingConfig {
+
+    public static final boolean DEFAULT_LOG_INPUT_DETAILS = true;
+    public static final boolean DEFAULT_LOG_INPUT_COLLECTION_AWARE = false;
+    public static final boolean DEFAULT_LOG_RESULT_DETAILS = true;
+    public static final boolean DEFAULT_LOG_RESULT_COLLECTION_AWARE = false;
+
+    @JsonProperty("service")
+    private List<LogConfiguration> serviceLoggingConfigs;
+
+    @JsonProperty("api")
+    private List<LogConfiguration> apiLoggingConfigs;
+
+    @JsonProperty("lep")
+    private List<LepLogConfiguration> lepLogConfigurations;
+
+    public Map<String, LogConfiguration> buildServiceLoggingConfigs() {
+        return buildLogConfiguration(this.getServiceLoggingConfigs());
+    }
+
+    public Map<String, LogConfiguration> buildApiLoggingConfigs() {
+        return buildLogConfiguration(this.getApiLoggingConfigs());
+    }
+
+    public Map<String, LepLogConfiguration> buildLepLoggingConfigs(String tenantKey) {
+        Map<String, LepLogConfiguration> configs = new HashMap<>();
+        if (CollectionUtils.isEmpty(lepLogConfigurations)) {
+            return configs;
+        }
+        lepLogConfigurations.forEach(lepLogConfiguration -> {
+            String pathKey = "lep://" + tenantKey.toUpperCase() + "/"
+                + lepLogConfiguration.getGroup() + "/"
+                + lepLogConfiguration.getFileName();
+
+            configs.put(pathKey, lepLogConfiguration);
+        });
+        return configs;
+    }
+
+    private Map<String, LogConfiguration> buildLogConfiguration(List<LogConfiguration> logConfigurations) {
+        Map<String, LogConfiguration> configs = new HashMap<>();
+        if (CollectionUtils.isEmpty(logConfigurations)) {
+            return configs;
+        }
+
+        logConfigurations.forEach(it -> {
```

Comment:
can we separate this to 3 filter and collect(toMap({}, {}))?

Reply:
added toMap() and buildConfigKey method